### PR TITLE
Multi-threaded logging

### PR DESCRIPTION
**Pull request by ghost**
Uses Mutex within the spin crate in order to make this logging multi-threaded.

### CODE REVIEW

**Comment by ghost** [comment]
big commits number lol

**Comment by YushiOMOTE** [comment]
Thank you for the PR!!

> big commits number lol

Would you mind if I squash on merge?

File: src/logger.rs [hunk 1]
```diff
@@ -1,34 +1,30 @@
...
             format_args!(
-                "{:>8}: {} ({}, {}:{})\n",
+                "[{}  {}] {}\n\0",
                 record.level(),
-                record.args(),
-                record.target(),
-                record.file().unwrap_or("<unknown>"),
-                record.line().unwrap_or(0),
+                record.module_path().unwrap_or(record.target()),
+                record.args()
```
Comment (YushiOMOTE) [inline src/logger.rs:27]: Why is this format change required?

File: Cargo.toml [hunk 2]
```diff
@@ -11,5 +11,6 @@ license = "MIT"
 readme = "README.md"
 
 [dependencies]
-uart_16550 = "0.2"
-log = { version = "0.4", default-features = false }
+uart_16550 = "0.3.0"
```
Comment (YushiOMOTE) [inline Cargo.toml:14]: :+1:

File: src/logger.rs (hunk 1 above)
Comment (ghost) [reply src/logger.rs:27]: no its not required, it can be discarded

**Comment by ghost** [comment]
yes you can squash

File: src/logger.rs (hunk 1 above)
Comment (YushiOMOTE) [reply src/logger.rs:27]: Can you discard the format change part? The other part LGTM 👍

Comment (YushiOMOTE) [reply src/logger.rs:27]: @Azvanzed Can you discard the format change part? The other part LGTM 👍

Comment (YushiOMOTE) [reply src/logger.rs:27]: Let me handle this :+1:

**Pull request merged**